HeterogeneousCore/AlpakaInterface/interface/getDeviceCachingAllocator.h

```diff
@@ -60,11 +60,11 @@ namespace cms::alpakatools {
       }
 
       // use a custom deleter to destroy all objects and deallocate the memory
-      auto deleter = [size](Allocator* ptr) {
+      auto deleter = [size](Allocator* lptr) {
         for (size_t i = size; i > 0; --i) {
-          std::destroy_at(ptr + i - 1);
+          std::destroy_at(lptr + i - 1);
         }
-        std::allocator<Allocator>().deallocate(ptr, size);
+        std::allocator<Allocator>().deallocate(lptr, size);
```

Comment:
I disagree that `ptr` inside the lambda may shadow `ptr` in the enclosing function, since it is not captured.

Reply:
Here is a simple reproducer:
```c++
#include <iostream>

int main(void) {
  std::cout << std::boolalpha;

  bool x = false;
  std::cout << "outside: " << x << std::endl;

  [](){
    bool x = true;
    std::cout << "inside: " << x << std::endl;
  }();

}
```

Compiling it with `-Wshadow` results in
```
test.cc: In lambda function:
test.cc:10:10: warning: declaration of ‘x’ shadows a previous local [-Wshadow]
   10 |     bool x = true;
      |          ^
test.cc:6:8: note: shadowed declaration is here
    6 |   bool x = false;
      |        ^
```

This is a false positive. In fact, commenting out the declaration inside the lambda:
```diff
   [](){
-    bool x = true;
+    //bool x = true;
     std::cout << "inside: " << x << std::endl;
   }();
```
results in a hard error:
```
test.cc: In lambda function:
test.cc:11:32: error: ‘x’ is not captured
   11 |     std::cout << "inside: " << x << std::endl;
      |                                ^
```

Reply:
In general, I'm not particularly fond of having to modify valid code to keep the compiler happy - and even less so for false positives.

Reply:
there is an example in https://godbolt.org/z/5b6Ybf136 from
https://stackoverflow.com/questions/66404751/variable-shadowed-warning-in-lambda-when-not-captured
would it not apply here?

I recall that a solution we often used for other false-positives was to tell the compiler that this an OK code.

Reply:
> there is an example in https://godbolt.org/z/5b6Ybf136 from

uhm, on a second thought, it doesn't look like a good example

Reply:
curiously enough, this place started this PR, since the compiler polluted our project (LST) compilation. We want to use `Wshadow` to help improve the code.
If the proposed solution doesn't work for heterogeneous, please propose an alternative that can disable the warning.

Reply:
> ```c++
> #pragma GCC diagnostic ignored "-Wshadow"
> ```

thank you for the suggestion; updated now

admittedly, the initial 3-character edit was less imposing.